docs/proposals/scheduling/20220530-fine-grained-cpu-orchestration.md

```diff
@@ -0,0 +1,690 @@
+
+##### CRD Scheme definition
+
+We use [NodeResourceTopology](https://github.com/k8stopologyawareschedwg/noderesourcetopology-api/blob/master/pkg/apis/topology/v1alpha1/types.go) CRD to describe the NUMA Topology. The community-defined NodeResourceTopology CRD is mainly used for the following considerations:
+
+- NodeResourceTopology already contains basic NUMA topology information and kubelet TopologyManager's Scope and Policies information. We can reuse the existing codes.
+- Keep up with the evolution of the community and influence the community to make more changes.
+
+##### Compatible
+
+In order to be compatible with the existing NodeResourceTopology instances and prevent koordlet from conflicting with existing components, when koordlet creates a NodeResourceTopology, add the prefix `koord-` before the name to distinguish it, and add the label `node.koordinator.sh/managed-by=true` describes the node is managed by Koordinator.
+
+##### Extension
+
+At present, the NodeResourceTopology lacks some information, and it is temporarily written in the NodeResourceTopology in the form of annotations or labels:
+
+- The annotation `kubelet.koordinator.sh/cpu-manger-policy` describes the kubelet CPU manager policy and options. The scheme is defined as follows:
+
+```go
+const (
+  FullPCPUsOnlyOption            string = "full-pcpus-only"
+  DistributeCPUsAcrossNUMAOption string = "distribute-cpus-across-numa"
+)
+
+type KubeletCPUManagerPolicy struct {
+  Policy  string            `json:"policy,omitempty"`
+  Options map[string]string `json:"options,omitempty"`
+}
+
+```
+
+- The annotation `node.koordinator.sh/cpu-topology` describes the detailed CPU topology. Fine-grained management mechanism needs more detailed CPU topology information. The scheme is defined as follows:
+
+```go
+type CPUTopology struct {
+  Detail []CPUInfo `json:"detail,omitempty"`
+}
+
+type CPUInfo struct {
+  CPUID        int32 `json:"cpu"`
+  CoreID       int32 `json:"core"`
+  NUMASocketID int32 `json:"numaSocket"`
+  NUMANodeID   int32 `json:"numaNode"`
+}
+```
+
+- annotation `node.koordinator.sh/pod-cpu-allocs` describes the CPUs allocated by Koordinator LSE/LSR and K8s Guaranteed Pods. The scheme corresponding to the annotation value is defined as follows:
+
+```go
+type PodCPUAlloc struct {
+  Namespace string    `json:"namespace,omitempty"`
+  Name      string    `json:"name,omitempty"`
+  UID       types.UID `json:"uid,omitempty"`
+  CPU       string    `json:"cpu,omitempty"`
+}
+
+type PodCPUAllocs []PodCPUAlloc
+```
+
+- The annotation `node.koordinator.sh/cpu-shared-pools` describes the CPU Shared Pool defined by Koordinator. The shared pool is mainly used by Koordinator LS Pods or K8s Burstable Pods. The scheme is defined as follows:
+
+```go
+type NUMACPUSharedPools []CPUSharedPool
+
+type CPUSharedPool struct {
+  NUMASocketID int32  `json:"numaSocket"`
```

Comment:
SocketID
NodeId

remove Numa prefix will be more clear

Reply:
done